watcher/src/watcher.rs

```diff
@@ -215,8 +219,8 @@ impl Watcher {
             }
 
             // If nothing succeeded, maybe we are synced all the way through or something
-            // else is wrong.
-            if !had_success {
+            // else is wrong. Or if we've reached max blocks per iteration, exit.
+            if !had_success || counter > MAX_BLOCKS_PER_SYNC_ITERATION {
```

Comment:
I think this is suboptimal - it will result the code in https://github.com/mobilecoinfoundation/mobilecoin/blob/master/watcher/src/bin/main.rs#L150-L159 taking longer to execute since there will be more sleeps in between each batch.

The default poll interval is 1 second, so this would result in more than extra 250 seconds when syncing the mainnet ledger.

Reply:
This point is true. Thus I refactored towards your enum suggestion with the small change below. Let me know what you think.

`enum SyncLimit {
    NoLimit,
    MaxBlockHeight{ max_block_height: u64, max_blocks_to_sync: u32 },
 }
`